plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -306,31 +310,50 @@ protected void renameTable(ConnectorSession session, String catalogName, String
             throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables across schemas");
         }
 
-        super.renameTable(session, catalogName, schemaName, tableName, newTable);
-    }
-
-    @Override
-    protected String renameTableSql(String catalogName, String remoteSchemaName, String remoteTableName, String newRemoteSchemaName, String newRemoteTableName)
-    {
-        return format(
-                "sp_rename %s, %s",
-                singleQuote(catalogName, remoteSchemaName, remoteTableName),
-                singleQuote(newRemoteTableName));
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String newSchemaName = newTable.getSchemaName();
+            String newTableName = newTable.getTableName();
+            verifyTableName(connection.getMetaData(), newTableName);
+            ConnectorIdentity identity = session.getIdentity();
+            String newRemoteSchemaName = identifierMapping.toRemoteSchemaName(identity, connection, newSchemaName);
+            String newRemoteTableName = identifierMapping.toRemoteTableName(identity, connection, newRemoteSchemaName, newTableName);
+
+            String fullTableFromName = DOT_JOINER.join(catalogName, schemaName, quoteEscapeForSqlServer(tableName));
+
+            try (CallableStatement renameTable = connection.prepareCall("exec sp_rename ?, ?")) {
+                renameTable.setString(1, fullTableFromName);
+                renameTable.setString(2, newRemoteTableName);
+                renameTable.execute();
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
     }
 
     @Override
-    protected String renameColumnSql(JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newRemoteColumnName)
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
     {
         RemoteTableName remoteTableName = handle.asPlainTable().getRemoteTableName();
-        return format(
-                "sp_rename %s, %s, 'COLUMN'",
-                singleQuote(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), "[" + escape(jdbcColumn.getColumnName()) + "]"),
-                "[" + newRemoteColumnName + "]");
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String columnFrom = DOT_JOINER.join(remoteTableName.getCatalogName().orElseThrow(), remoteTableName.getSchemaName().orElseThrow(), quoteEscapeForSqlServer(remoteTableName.getTableName()), quoteEscapeForSqlServer(jdbcColumn.getColumnName()));
```

Comment:
why `orElseThrow`? is it not possible to have them `null`?

Same question here - deserves a comment why only table and column name need quoting?

Reply:
to be honest I didn't check special character in schema name, because we didn't support schema renaming in SQL Server:
```
  @Override
    public void renameSchema(ConnectorSession session, String schemaName, String newSchemaName)
    {
        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
    }
```

Reply:
Good catch, thx, SQL server supports tricky schema names so if we quote with the same method schema names we will be able to rename tables and columns

Reply:
> why `orElseThrow`? is it not possible to have them `null`?

I think no, we should not process such cases

Reply:
> why `orElseThrow`? is it not possible to have them `null`?

@hashhar depends on the remote database .  eg some have catalogs and some do not

Reply:
.. and here we know the database, so we know what's expected